### DISCUSSION TITLE
Icon Suggestions

### DISCUSSION THREAD

**Discussion by JeremyGrieshop** [tailwindlabs/heroicons#377] [Ideas]
I really love the work that you've done so far!  If you need any more suggestions, I'm currently working on a fairly large (soon to be open source) project that requires very data-centric icons.   

Common editor things like:
- cut, save, save as, discard, files, convert

Software (very few icon libraries have these):
 - objects, functions, variables, expressions, data types, engine/core/kernel, extensions

Databases:
 - flat, structured, spreadsheet, object, directory, query

**Comment by cotiga** [comment]
Great job, i think it's missing
open eye
closed eye
Thanks

**Comment by Jam0r85** [comment]
Can we have a restore as well please!

**Comment by samuei** [comment]
A couple of those are already in here, like [download](https://github.com/refactoringui/heroicons/blob/master/src/outline-md/md-download.svg), [upload](https://github.com/refactoringui/heroicons/blob/master/src/outline-md/md-upload.svg), [filter](https://github.com/refactoringui/heroicons/blob/master/src/outline-md/md-filter.svg), and [source code](https://github.com/refactoringui/heroicons/blob/master/src/outline-md/md-code.svg).

**Comment by JeremyGrieshop** [comment]
> A couple of those are already in here, like [download](https://github.com/refactoringui/heroicons/blob/master/src/outline-md/md-download.svg), [upload](https://github.com/refactoringui/heroicons/blob/master/src/outline-md/md-upload.svg), [filter](https://github.com/refactoringui/heroicons/blob/master/src/outline-md/md-filter.svg), and [source code](https://github.com/refactoringui/heroicons/blob/master/src/outline-md/md-code.svg).

Thank you, I didn't see those.  Comment edited.

**Comment by compufox** [comment]
a "log out" icon would be really useful as well

**Comment by epixian** [comment]
> a "log out" icon would be really useful as well

logout is included as of https://github.com/refactoringui/heroicons/commit/41bd40f1bd605dd1d71cf2a8b22c6c772c284d12

**Comment by ryanpurcella** [comment]
Education hat icon might  be nice

**Comment by geekish** [comment]
Definitely would love a classic floppy disk save icon.
